.github/workflows/build-deploy-dev.yml

```diff
@@ -125,23 +125,22 @@ jobs:
       - name: Log in to OpenShift
         run: |
           oc login --token=${{ secrets.SA_TOKEN }} --server=https://api.silver.devops.gov.bc.ca:6443
-      # Build the images
-      - name: Build SIMS-API Image
+          if [ $? -ne 0 ]; then
+            echo "Failed to log in to OpenShift. Please check the credentials and connection details."
+            exit 1
+          fi
+       # Build the images in parallel.
+      - name: Build db-migrations, api, workers, queues and web Images in parallel
         working-directory: "./devops"
         run: |
-          make oc-build-api
-      - name: Build Workers Image
-        working-directory: "./devops"
-        run: |
-          make oc-build-workers
-      - name: Build Queue Consumers Image
-        working-directory: "./devops"
-        run: |
-          make oc-build-queue-consumers
-      - name: Build Web/Frontend Image
-        working-directory: "./devops"
+            make oc-build-db-migrations &
```

Comment:
Why this approach was taken instead of using the steps as they were before? 
The benefit from the previous way was also to have better UI process descriptions on Github actions, as shown below.
![image](https://github.com/bcgov/SIMS/assets/61259237/b96ed13d-3b6d-482a-bd65-0b27df9aa4c2)

We also have other examples about how to run jobs in parallel, for instance on file `tests.yml`, and also how to run in parallel and create a dependency where one job must be finished to another start, for instance on the file `deploy-all.yml` as shown below.
![image](https://github.com/bcgov/SIMS/assets/61259237/746a9ab6-549d-4181-88fc-c734427d9dee)

If there are benefits in the approach used in this PR, can you please clarify them?

Reply:
Both are similar approaches, but the choice I made provides additional checks and error handling to ensure that the database migrations are successfully executed and that all the parallel deployments complete before continuing with the workflow. It enhances the reliability of the deployment process by explicitly handling potential failures and ensuring proper synchronization.
Also the approach on handing the build and deployments as a single line, is just an option, I can revert it back, but we are not going to lose any logs and start and end of the make targets, using this approach and its descriptive in knowing that the jobs run in parallel.

Reply:
Thanks for doing the changes with the parallel GitHub actions as per the Team discussions 😉